fiona/ogrext.pyx

```diff
@@ -1017,6 +1022,20 @@ cdef class WritingSession(Session):
                 raise RuntimeError("Failed to write record: %s" % record)
             _deleteOgrFeature(cogr_feature)
 
+            features_in_transaction += 1
+            if features_in_transaction == 20000:  # TODO: this shouldn't be hard-coded
```

Comment:
@snorfalorpagus how about this for a compromise: until we make transactions a Fiona software feature, say in 1.8, let's use a constant value here. `DEFAULT_TRANSACTION_SIZE = 20000` or something like that defined in the module. Then we'd have the same performance characteristics as ogr2ogr in many cases and users would have two work-arounds for different transaction sizes:

1. a user could monkey patch the constant in code `fiona.ogrext.DEFAULT_TRANSACTION_SIZE = 1000`.
2. better: a user could call `writerecords()` with batches smaller than 20,000 and each of these would be done in their own transaction, right?